### PR TITLE
Remove binary channel access to silicons without laws.

### PR DESCRIPTION
**Pull request by VasilisThePikachu**
## About the PR

Remove binary channel access to silicons without laws.

## Why / Balance

Mostly feature proofing but these should not have binary access. ESPECIALLY pais

Say we had malf ai and it was talking with other borgs about its plans. The owner of the pai (or any of these silicons) can just listen into this. Someone can just wake up a pai and ask it to monitor the binary channel. There's a reason the binary radio chip is a syndicate item.

Some of these also have no reason to be able to talk on binary, are you gonna ask the ai to bother someone to refill your vend stock? Are you gonna clown with the AI and other borgs as a clown borg and annoy them?

And the pai story can be repeated again, free binary access key.

## Requirements

- [x] I have read and am following the [Pull Request and Changelog Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html).
- [x] I have added media to this PR or it does not require an ingame showcase.

## Breaking changes

MobSiliconBase no longer has a radio. It will need to be added to anything requiring it. Anything requiring it already has it added

**Changelog**

:cl: Myra
- remove: Silicon with no laws will not have binary channel access.

### CODE REVIEW

**Comment by ScarKy0** [comment]
Me when my owner dies and im forced to just sit there in silence or /ghost

**Comment by Ilya246** [comment]
i think all the things having binary channel adds to fun more than it offsets balance so it should be kept, and makes being said silicons at least somewhat less boring plus allow you to communicate remotely

from my own experience, pAI-borg binary channel interactions add quite a bit to fun and general interestingness of gameplay

**Comment by deltanedas** [comment]
pais having binary would make malf impossible since any kind of order would immediately be snitched on and sec would raid the ai core

**Comment by Ilya246** [comment]
> pais having binary would make malf impossible since any kind of order would immediately be snitched on and sec would raid the ai core

then make all of those also obey ai

**Review by PJB3005 — APPROVED** [review]

**Comment by Simyon264** [comment]
I'd merge this once Malf AI is actually implemented. Currently cyborgs along with all other silicons act mostly independent of each other. Having played pAI a lot, there is a lot of fun to be had conversing with other pAIs and cyborgs. Being locked **only** to local chat sucks a lot and if this is merged pAI's need some sort of radio to continue being as fun as before.

**Comment by Golinth** [comment]
I wholeheartedly agree with this. pAIs already make stealing PDAs or killing people far more risky than it should be (zero counterplay if the pAI stays quiet until they report you), and when malf AI comes it will be even worse. As much as it “makes sense” for machines to have access to the “machine chat”, for balance purposes they need to have it revoked

**Comment by PJB3005** [comment]
> I'd merge this once Malf AI is actually implemented.

image

It is. There is no reason to have this feature other than pissing on 20 years of SS13 hindsight.

**Comment by Simyon264** [comment]
> > I'd merge this once Malf AI is actually implemented.
> 
> image
> 
> It is. There is no reason to have this feature other than pissing on 20 years of SS13 hindsight.

IIRC that board is not mapped anywhere currently. Also, malf AI has no real implementation beyond laws. There are no win conditions like with (as an example) nukies. A Malf AI cannot do anything to protect its laws from being changed again. One engineer is all it takes.

**Comment by PJB3005** [comment]
> IIRC that board is not mapped anywhere currently. Also, malf AI has no real implementation beyond laws. There are no win conditions like with (as an example) nukies. A Malf AI cannot do anything to protect its laws from being changed again. One engineer is all it takes.

I am sorry but this entire comment just does not understand how AI works in SS13. Just because "Malf AI" isn't a gamemode does not mean a subverted/hostile AI is not a game mechanic. It also does not matter how easily the AI can be reset, it can still do damage to players and the station in the mean time.

**Comment by Simyon264** [comment]
> not mean a subverted/hostile AI is not a game mechanic

It's a poorly implemented game mechanic for Station AI and limiting the binary channel in the current implementations will result in less RP and Gameplay; An overall worse experience.

**Comment by ArcaneOcto** [comment]
I think that these roles should have laws causing them to be loyal to the AI or something. I personally don't think removing the ability for pAIs to use Binary is a good idea. It's one of the few actually useful things you can do as a pAI. The AI and Borgs are arguably the most informed beings on the station, so being able to talk to them is very helpful to keep your owner informed. pAIs snitching on malf AI is definately something that needs to be fixed though.

**Comment by southbridge-fur** [comment]
I want to echo what a lot of people are saying regarding how pAIs having binary access is pretty much one of their only features. I want to follow that up by adding that there is another human there, playing the pAI. If I'm the target of an antag or join late in observer I'd like to be able to have some kind of influence on the game. Even if that influence is basically a middle-man for the human I'm attached to I still get to have some fun. 

I do agree though with the ideas I've seen in this and other PRs that add laws to pAIs, since that can add more dynamic interactions throughout a round.

**Comment by superjj18** [comment]
Let's be real here this change is not a matter of if but when

**Comment by nymhole** [comment]
opposite, ADD LAWS TO PAI.

**Comment by superjj18** [comment]
> opposite, ADD LAWS TO PAI.

Already denied by maintainers

**Comment by PJB3005** [comment]
Giving pAIs laws would literally go completely against the point of pAIs. They're supposed to be ***personal*** companions of a human, not servants of the station AI. Imagine your pAI you've been buddies with all shift having to shut up because the AI got subverted.

**Comment by ItsMeThom** [comment]
how about giving pAIs a slot for installing comms keys, like a headset? So pAIs could be given say, a cargo or sec key by their owner and allowed to communicate over the frequencies with/for their owner?

**Comment by slarticodefast** [comment]
This was brought up for maintainer vote and the unanimous result was to merge this as it is needed for the malf AI gamemode in the future.

**Comment by squirrelanna** [comment]
If the main contention is that pAIs having binary hurts the malf AI in particular, why not have a comms channel that also includes all networked electronic beings and devices? One that can also be more easily accessed by crew members? It feels seriously awful when you get isolated in an RP-based multiplayer game through no control of your own, and having SOME way to communicate at least gives you a chance to stay in the round and have interaction even if you're just lost in space. This will allow binary to become a way for antagonist silicons to communicate and preserve the current functionality of pAIs. Or just allow encryption keys and the ability to speak on those comms.

**Comment by VasilisThePikachu** [comment]
> If the main contention is that pAIs having binary hurts the malf AI in particular, why not have a comms channel that also includes all networked electronic beings and devices? One that can also be more easily accessed by crew members? It feels seriously awful when you get isolated in an RP-based multiplayer game through no control of your own, and having SOME way to communicate at least gives you a chance to stay in the round and have interaction even if you're just lost in space. This will allow binary to become a way for antagonist silicons to communicate and preserve the current functionality of pAIs. Or just allow encryption keys and the ability to speak on those comms.

This is what happens in some ss13 servers from what I was told, but it's not part of this PR. I would not mind seeing a PR adding it

**Comment by LordEclipse** [comment]
> If the main contention is that pAIs having binary hurts the malf AI in particular, why not have a comms channel that also includes all networked electronic beings and devices? One that can also be more easily accessed by crew members? It feels seriously awful when you get isolated in an RP-based multiplayer game through no control of your own, and having SOME way to communicate at least gives you a chance to stay in the round and have interaction even if you're just lost in space. This will allow binary to become a way for antagonist silicons to communicate and preserve the current functionality of pAIs. Or just allow encryption keys and the ability to speak on those comms.

Isn't that just Common?

**Comment by ScarKy0** [comment]
> Isn't that just Common?

IIRC pAI etc can't speak on Common?

**Comment by slarticodefast** [comment]
> IIRC pAI etc can't speak on Common?

For pAI the reason they cannot speak on common is that they should not be able to should "help, my owner is being attacked" on the radio the moment an antagonist is making a move. This would be very frustrating as an antag if your target has a pAI with them and you try to be stealthy. There have been PRs to give pAIs the ability to speak on common and they have been refused for that reason. And the same thing is a problem with them being able to speak on binary as well.

**Comment by LordEclipse** [comment]
> > Isn't that just Common?
> 
> IIRC pAI etc can't speak on Common?

Then let them

**Comment by ScarKy0** [comment]
> And the same thing is a problem with them being able to speak on binary as well.

pAI gameplay gonna get very boring bros 😭 
Chatting with silicons was hella fun

**Comment by Welldzhik** [comment]
> > IIRC pAI etc can't speak on Common?
> 
> For pAI the reason they cannot speak on common is that they should not be able to shout "help, my owner is being attacked" on the radio the moment an antagonist is making a move. This would be very frustrating as an antag if your target has a pAI with them and you try to be stealthy. There have been PRs to give pAIs the ability to speak on common and they have been refused for that reason. And the same thing is a problem with them being able to speak on binary as well.

What about just make a communication jammer cheaper? Two times like that. I mean, if you make it available, this problem becomes irrelevant. And not that it could somehow break the balance. Nobody uses it. In my memory, at least. For 4 TK you can buy a full-fledged weapon or tool.

**Comment by ArcaneOcto** [comment]
> > pAI gameplay gonna get very boring bros 😭 Chatting with silicons was hella fun

Hopefully pAIs can get laws later on and get this functionality again🤞
I think all the pAIs turning evil in the case of a malf AI would be sick.

**Comment by ArtisticRoomba** [comment]
Maybe when we get frequency channels on radios we can have a separate "Widenet" channel for all silicon machines to talk, while keeping "binary" separate for Malf AI purposes. I'd have to look and see what SS13 did in terms of keeping pAIs/borgs separate.

**Comment by Lucidueye** [comment]
Malf AI isn't in he game yet, as it stands this is just screwing over players. There's practically nothing to do as a pAI without the ability to speak on binary other than be a glorified map and mp3 player. 
If you owner is mute or a mime and getting attacked, then there's nothing you can do. You can't get help for them and they can't call for help on their own.
When Malf AI is implented it can have its own channel to use with borgs to prevent spying, or all silicons can be assigned to a different channel as suggested [here](https://github.com/space-wizards/space-station-14/pull/32385#issuecomment-2377792637)

**Comment by LordEclipse** [comment]
Maybe encryption keys can be slotted into pAIs? I wonder how that would go.

**Comment by VasilisThePikachu** [comment]
Please continue discussion in a development channel on discord or on the forums. Thank you

**Pull request merged**
